src/util/sss_regexp.h

```diff
@@ -0,0 +1,103 @@
+/*
+    SSSD
+
+    Authors:
+        Tomas Halman <thalman@redhat.com>
+
+    Copyright (C) 2018 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SSS_REGEXP_H_
+#define SSS_REGEXP_H_
+
+#include <stddef.h>
+#include <talloc.h>
+#include "config.h"
+
+/* regexp class */
+typedef struct _sss_regexp_t sss_regexp_t;
```

Comment:
This is again more of a general comment. Typically, we try to avoid typedefs for struct in SSSD - if it's a struct, let's call it that. But here I actually think it makes sense, because this type is an 'alias' for the previous pcre type.

I'm mostly bringing it up explicitly in case other developers are reading the review comments are want to chime in with their opinion :)